src/angular/components/sso.component.ts

```diff
@@ -53,13 +53,32 @@ export class SsoComponent {
                 if (qParams.code != null && codeVerifier != null && state != null && state === qParams.state) {
                     await this.logIn(qParams.code, codeVerifier);
                 }
+                const clientId = await this.storageService.get<string>(ConstantsService.ssoClientId);
+                if (clientId === ConstantsService.browserClientId) {
+                    // Prevent entering this if we're not in the web vault.
+                    if (location.protocol === 'https://') {
+                        window.postMessage({ type: "AUTH_RESULT", code: qParams.code, state: qParams.state }, "*");
+                        this.router.navigate(['sso-complete']);
+                    }
+                }
             } else if (qParams.clientId != null && qParams.redirectUri != null && qParams.state != null &&
                 qParams.codeChallenge != null) {
                 this.redirectUri = qParams.redirectUri;
                 this.state = qParams.state;
                 this.codeChallenge = qParams.codeChallenge;
                 this.clientId = qParams.clientId;
             }
+
+            if (qParams.clientId == ConstantsService.browserClientId) {
+                this.redirectUri = qParams.redirectUri;
+                this.state = qParams.state;
+                this.codeChallenge = qParams.codeChallenge;
+                this.clientId = qParams.clientId;
+                await this.storageService.save(ConstantsService.ssoCodeVerifierKey, qParams.codeVerifier);
+                await this.storageService.save(ConstantsService.ssoStateKey, qParams.state);
```

Comment:
I cannot think of a use case where these need to be persisted in the sso component like this. This should be done on the client itself, the browser extension whenever they click the button on the login/home page.